Sources/ZIPFoundation/FileManager+ZIP.swift

```diff
@@ -98,16 +98,7 @@ extension FileManager {
         guard let archive = Archive(url: sourceURL, accessMode: .read, preferredEncoding: preferredEncoding) else {
             throw Archive.ArchiveError.unreadableArchive
         }
-        // Defer extraction of symlinks until all files & directories have been created.
-        // This is necessary because we can't create links to files that haven't been created yet.
-        let sortedEntries = archive.sorted { (left, right) -> Bool in
-            switch (left.type, right.type) {
-            case (.directory, .file): return true
-            case (.directory, .symlink): return true
-            case (.file, .symlink): return true
```

Comment:
Adding an extra case to unpack symlinks in the order they are in the archive (`case (.symlink, .symlink): return true` also fixes the issue, but I suppose the proposed solution is more thorough and will cover cases where the symlinks are not in the correct order.